### PR TITLE
Move supported site K8s version to shortcodes

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation

> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

Implements two new shortcodes:
* {{% k8s-version %}} - which outputs the currently supported version
* {{% k8s-api href="#podtemplatespec-v1-core" %}} - which outputs the api reference url to the supported k8s version.

These shortcode utilise the `HUGO_ENV` environment to determine if it should show the current K8s version, or the next - as it is only set to the value of "production" when it is generated for the release version of the agones.dev website.

Also added updates to the release checklist to manage this as well.

Ideally, this will remove lots of busy work of feature shortcoding a lot of content as we churn through Kubernetes versions.

**Which issue(s) this PR fixes**:

Work on #1824

**Special notes for your reviewer**:

Long term, we may want to expand this to include separate tools for current and next full semver versions - i.e. rather than just
1.16 ➡ 1.17, but something like 1.16.3 ➡ 1.17.14.

We cheat a little on this release as there is 1.16.13 and 1.17.13 released on both 1.16 and 1.17.

### CODE REVIEW

**Comment by markmandel** [comment]
I'm kinda proud of this one - should hopefully save lots of time in the long run :smile:

File: site/content/en/docs/Installation/Creating Cluster/aks.md [hunk 1]
```diff
@@ -28,8 +28,8 @@ az group create --name $AKS_RESOURCE_GROUP --location $AKS_LOCATION
 
 # Create the AKS cluster - this might take some time. Type 'az aks create -h' to see all available options
 
-# The following command will create a four Node AKS cluster. Node size is Standard A1 v1 and Kubernetes version is 1.16.13. Plus, SSH keys will be generated for you, use --ssh-key-value to provide your values
-az aks create --resource-group $AKS_RESOURCE_GROUP --name $AKS_NAME --node-count 4 --generate-ssh-keys --node-vm-size Standard_A4_v2 --kubernetes-version 1.16.13
+# The following command will create a four Node AKS cluster. Node size is Standard A1 v1 and Kubernetes version is {{% k8s-version %}}.13. Plus, SSH keys will be generated for you, use --ssh-key-value to provide your values
```
Comment (roberthbailey) [inline site/content/en/docs/Installation/Creating Cluster/aks.md:31]: We need to remember to change the minor versions too. Maybe make a short-code for aks-k8s-minor-version?

File: site/content/en/docs/Installation/Creating Cluster/minikube.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ The following command starts a local minikube cluster via virtualbox - but this
 replaced by a [vm-driver](https://github.com/kubernetes/minikube#requirements) of your choice.
 
 ```bash
-minikube start --kubernetes-version v1.16.13 --vm-driver virtualbox
+minikube start --kubernetes-version v{{% k8s-version %}}.13 --vm-driver virtualbox
```
Comment (roberthbailey) [inline site/content/en/docs/Installation/Creating Cluster/minikube.md:33]: minikube-k8s-minor-version

Comment (markmandel) [reply site/content/en/docs/Installation/Creating Cluster/minikube.md:33]: These are good ideas. I'll make this change :+1:

Comment (markmandel) [reply site/content/en/docs/Installation/Creating Cluster/minikube.md]: Done! I like it!

**Review by roberthbailey — APPROVED** [review]
Not sure why the tests failed, but this looks great. It will definitely decrease our maintenance for each future k8s version upgrade. :)

**Pull request merged**